de.dlr.sc.overtarget.language/src/de/dlr/sc/overtarget/language/generator/util/ReferenceTargetHelper.xtend

```diff
@@ -0,0 +1,105 @@
+/*******************************************************************************
+ * Copyright (c) 2020 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package de.dlr.sc.overtarget.language.generator.util
+
+import de.dlr.sc.overtarget.language.targetmodel.TargetModel
+import de.dlr.sc.overtarget.language.targetmodel.TargetFile
+import de.dlr.sc.overtarget.language.targetmodel.RepositoryLocation
+import org.eclipse.emf.ecore.util.EcoreUtil
+import java.util.ArrayList
+import org.eclipse.core.resources.ResourcesPlugin
+import de.dlr.sc.overtarget.language.util.TargetPlatformHelper
+import org.eclipse.core.resources.IFile
+import org.eclipse.core.runtime.Path
+
+/**
+ * This class processes the model data for generation
+ */
+class ReferenceTargetHelper {
+
+	def static getModelToGenerate(TargetModel model) {
+		for (RepositoryLocation repos : model.repositoryLocations) {
+			if (repos.referenceTarget == true) {
+				val list = newArrayList
+				for (RepositoryLocation repos2 : model.repositoryLocations) {
```

Comment:
I don't quite get what's happening here... Maybe add some comments and don't class an attribute **2... Rather name it after the difference to the first repo